src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs

```diff
@@ -159,12 +159,13 @@ public IEnumerable<QueryResult> Query(IndexQueryServerSide query, QueryTimingsSc
                             continue;
                         }
                         
-                        if (filter.TryFilter(document, key, _state, out var overLimit) == false)
+                        var filterResult = queryFilter?.Apply(document, key, _state);
```

Comment:
Better to use `switch` here, I think.

Reply:
`switch` is a bit problematic here because we want to `break` the for loop in case of `FilterResult.LimitReached`